loopingTimer.js

```diff
@@ -49,6 +49,17 @@ class LoopingTimer {
       }
     }
   }
+  
+  /**
+   * @returns {Duration} of time left in the current loop of the timer
+   */
+  getDuration() {
+    if (this.timer && !this.hasTerminated()) {
```

Comment:
Remove the `!hasTerminated` test. If the timer has terminated it means the loop has exited and this will return a negative duration telling us when the loop exited.